Assets/_BForBoss/_Entities/Enemies/FlyingRobot/Scripts/OffMeshLinkChecker.cs

```diff
@@ -0,0 +1,63 @@
+using UnityEngine;
+using UnityEngine.AI;
+
+namespace BForBoss
+{
+    public class OffMeshLinkChecker : MonoBehaviour
+    {
+        [SerializeField] private NavMeshAgent _agent = null;
+        [SerializeField, Tooltip("How much should the cost of a mesh link multiply by when in use?")]
+        private float _multiplier = 1000f;
+
+        private OffMeshLink _link = null;
+        private float _oldCost;
+
+        private void Update()
+        {
+            if (_agent.currentOffMeshLinkData.valid)
+            {
+                AcquireOffMeshLink();
+            }
+            else
+            {
+                ReleaseOffMeshLink();
+            }
+        }
+
+        private void AcquireOffMeshLink()
+        {
+            if (_link != null)
+            {
+                return;
+            }
+            
+            _link = _agent.currentOffMeshLinkData.offMeshLink;
+            _oldCost = _link.costOverride;
+            _link.costOverride = _oldCost <= 0.0f ? (_multiplier * 1) : _oldCost * _multiplier;
```

Comment:
`_oldCost <= 0.0f ? (_multiplier * 1) : _oldCost * _multiplier;` 

Double check this for any redundancies.

Reply:
I just copied this code straight from Gursaanj, but I can change the (_multiplier * 1) to just _multiplier if that is what you are referring to

Reply:
yup. If you see anything off, don't be afraid to update it.

Reply:
Sounds good! It has been changed 👍